lighthouse-core/gather/gather-runner.js

```diff
@@ -203,31 +203,34 @@ class GatherRunner {
   /**
    * Navigates to about:blank and calls beforePass() on gatherers before tracing
    * has started and before navigation to the target page.
-   * @param {!Object} options
+   * @param {!Object} passContext
    * @param {!GathererResults} gathererResults
    * @return {!Promise}
    */
-  static beforePass(options, gathererResults) {
-    const blockedUrls = (options.config.blockedUrlPatterns || [])
-      .concat(options.flags.blockedUrlPatterns || []);
-    const blankPage = options.config.blankPage;
+  static beforePass(passContext, gathererResults) {
+    const blockedUrls = (passContext.config.blockedUrlPatterns || [])
+      .concat(passContext.flags.blockedUrlPatterns || []);
+    const blankPage = passContext.config.blankPage;
 
     // On the very first pass we're already on blank
-    const skipLoadBlank = options.passIndex === 0;
-    let pass = skipLoadBlank
+    const skipLoadBlank = passContext.passIndex === 0;
+    const loadBlank = skipLoadBlank
       ? Promise.resolve()
-      : GatherRunner.loadBlank(options.driver, blankPage);
+      : GatherRunner.loadBlank(passContext.driver, blankPage);
 
     // Set request blocking before any network activity
     // No "clearing" is done at the end of the pass since blockUrlPatterns([]) will unset all if
     // neccessary at the beginning of the next pass.
-    pass = pass
-        .then(() => options.driver.blockUrlPatterns(blockedUrls))
-        .then(() => options.driver.setExtraHTTPHeaders(options.flags.extraHeaders));
+    const pass = loadBlank
+        .then(() => passContext.driver.blockUrlPatterns(blockedUrls))
+        .then(() => passContext.driver.setExtraHTTPHeaders(passContext.flags.extraHeaders));
 
-    return options.config.gatherers.reduce((chain, gatherer) => {
+    return passContext.config.gatherers.reduce((chain, gathererDefn) => {
       return chain.then(_ => {
-        const artifactPromise = Promise.resolve().then(_ => gatherer.beforePass(options));
+        const gatherer = gathererDefn.instance;
+        // Abuse the passContext to pass through gatherer options
+        passContext.options = gathererDefn.options || {};
+        const artifactPromise = Promise.resolve().then(_ => gatherer.beforePass(passContext));
```

Comment:
wdyt about a followup PR to rename `options` in all the gatherers to `passContext`?

Reply:
sg